### PR TITLE
Switch from functional to class losses

### PR DESCRIPTION
**Pull request by andrrizzi**
## Description
Changes the CV classes to use by default a loss function object.

It is still possible to assign ``loss_fn`` to a (lambda) function, although I had to override ``__setattr__`` to work around some weird PyTorch magic (hopefully the comments I left in the code makes it clear).

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by luigibonati** [comment]
thank you @andrrizzi! this sounds good to me, shall we merge it?

**Comment by andrrizzi** [comment]
Great, thanks! Merging then.
